protocol-units/bridge/service/tests/event_monitoring.rs

```diff
@@ -0,0 +1,68 @@
+use alloy::{
+	node_bindings::Anvil,
+	primitives::{address, keccak256},
+};
+use anyhow::Result;
+use bridge_service::chains::{
+	bridge_contracts::BridgeContract,
+	ethereum::types::{EthAddress, EthHash},
+};
+use bridge_service::chains::{
+	ethereum::{
+		client::{Config as EthConfig, EthClient},
+		event_monitoring::{Config as EthMonitoringConfig, EthMonitoring},
+	},
+	movement::{
+		client::{Config as MovementConfig, MovementClient},
+		event_monitoring::MovementMonitoring,
+	},
+};
+use tokio_stream::StreamExt;
+
+use bridge_service::types::{Amount, AssetType, BridgeAddress, HashLock};
+use harness::TestHarness;
+mod harness;
+mod utils;
+
+#[tokio::test]
+async fn test_should_receive_event() -> Result<()> {
+	let mut harness: TestHarness = TestHarness::new_only_eth().await;
+	let anvil = Anvil::new().port(harness.rpc_port()).spawn();
+
+	let signer_address = harness.set_eth_signer(anvil.keys()[0].clone());
+	harness.deploy_init_contracts().await;
+	let config = EthMonitoringConfig::default();
+	println!("config: {:?}", config);
+	let eth_stream: EthMonitoring = EthMonitoring::build(config).await?;
+
+	let eth_config = EthConfig::build_for_test();
+	let eth_client = EthClient::new(eth_config).await?;
+
+	let mvt_config = MovementConfig::build_for_test();
+	let mvt_client = MovementClient::new(&mvt_config).await?;
+
+	let mvt_stream = MovementMonitoring::build(mvt_config).await?;
+
+	//Start the relayer
+	bridge_service::run_bridge(eth_client, eth_stream, mvt_client, mvt_stream).await?;
```

Comment:
I think you should create the EthMonitoring and wait for the steam to detect the initiate transfer example.
Starting the bridge doesn't allow you to see if the event has been notified and processed by the EthMonitoring struct.